components/object_store/src/obkv/meta.rs

```diff
@@ -0,0 +1,303 @@
+        write!(f, "ObjectStore-Obkv-MetaManager({:?})", self.client)?;
+        Ok(())
+    }
+}
+
+impl<T: TableKv> MetaManager<T> {
+    pub async fn save_meta(&self, meta: ObkvObjectMeta) -> Result<()> {
+        let mut batch = T::WriteBatch::default();
+        let encode_bytes = meta.encode()?;
+        batch.insert(meta.location.as_bytes(), &encode_bytes);
+        self.client
+            .as_ref()
+            .write(WriteContext::default(), OBJECT_STORE_META, batch)
+            .box_err()
+            .with_context(|| SaveMeta {
+                location: meta.location,
+            })?;
+        Ok(())
+    }
+
+    pub async fn read_meta(&self, location: &Path) -> Result<Option<ObkvObjectMeta>> {
+        let value = self
+            .client
+            .as_ref()
+            .get(OBJECT_STORE_META, location.as_ref().as_bytes())
+            .box_err()
+            .context(ReadMeta {
+                location: location.as_ref().to_string(),
+            })?;
+
+        value.map(|v| decode_json(&v)).transpose()
+    }
+
+    pub async fn delete_meta(&self, meta: ObkvObjectMeta, location: &Path) -> Result<()> {
+        self.client
+            .as_ref()
+            .delete(OBJECT_STORE_META, location.as_ref().as_bytes())
+            .box_err()
+            .context(DeleteMeta {
+                location: meta.location,
+            })?;
+
+        Ok(())
+    }
+
+    pub async fn delete_meta_with_version(&self, location: &Path, version: &str) -> Result<()> {
+        let meta_result = self.read_meta(location).await?;
+        if let Some(meta) = meta_result {
+            if meta.version == version {
+                self.delete_meta(meta, location).await?;
+            }
+        }
+        Ok(())
+    }
+
+    pub async fn list_meta(
+        &self,
+        prefix: &Path,
+    ) -> StoreResult<Vec<ObkvObjectMeta>, std::io::Error> {
+        let scan_context: ScanContext = ScanContext {
+            timeout: time::Duration::from_secs(SCAN_TIMEOUT_SECS),
+            batch_size: SCAN_BATCH_SIZE,
+        };
+
+        let scan_request = util::scan_request_with_prefix(prefix.as_ref().as_bytes());
+
+        let mut iter = self
+            .client
+            .scan(scan_context, OBJECT_STORE_META, scan_request)
+            .map_err(|source| StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            })?;
+
+        let mut metas = vec![];
+        while iter.valid() {
+            let value = iter.value();
+            let meta = ObkvObjectMeta::decode(value).map_err(|source| StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            })?;
+            metas.push(meta);
+            iter.next().map_err(|source| StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            })?;
+        }
+        Ok(metas)
+    }
+}
+
+fn decode_json<'a, ObkvObjectMeta: serde::Deserialize<'a>>(
```

Comment:
Why not remove the two methods and put their contents just in the ObkvObjectMeta?